packages/api/src/controllers/stream.ts

```diff
@@ -1062,19 +1102,20 @@ function publishRecordingStartedHook(
     streamId: session.parentId,
     userId: session.userId,
     event: "recording.started",
-    payload: { session: toExternalSession(session, ingest) },
+    payload: { session: await toExternalSession(config, session, ingest) },
   });
 }
 
 /**
  * We don't actually send the webhook here, but schedule an event after a timeout.
  */
-function publishDelayedRecordingReadyHook(
+async function publishDelayedRecordingReadyHook(
```

Comment:
How much work do you think it would be to make the `recording.ready` hook only fire when the recording asset is created? Might end up with some convoluted dependencies between VOD and webhook flows, but maybe not (maybe that dep already exists after all). I don't think it needs to be included in this PR, but I think it would be a good followup to keep a sensible `recording.ready` webhook (that means what it says). WDYT?

Reply:
Good idea. Let's do it as a follow-up. Created a linear ticket: https://linear.app/livepeer/issue/VID-153/studio-trigger-webhook-only-if-recording-is-processed